include/range/v3/range_fwd.hpp

```diff
@@ -609,8 +601,16 @@ namespace ranges
             struct repeat_fn;
         }
 
-        template<typename Rng>
-        struct reverse_view;
+        namespace detail
+        {
+            template<typename>
+            std::false_type double_reverse(long);
+            template<typename R, meta::if_<std::is_same<R, typename R::unreverse_me>>* = nullptr>
+            std::true_type double_reverse(int);
+        }
+
+        template<typename Rng, bool = decltype(detail::double_reverse<Rng>(42))::value>
+        struct RANGES_EMPTY_BASES reverse_view;
```

Comment:
Boo. 👎

Reply:
Which part are you "boo"ing here?

Reply:
I don't think it makes sense for reversing a reverse view to be special, and if it were, I would expect a partial specialization of `reverse_view<reverse_view<...>>` to be sufficient. I don't like `unreverse_me`, and I don't like the extra `bool` template parameter.

Reply:
>  I would expect a partial specialization of reverse_view<reverse_view<...>> to be sufficient.

This was part of my distaste for [Option (3) from the `cycled_view` PR](https://github.com/ericniebler/range-v3/pull/923#discussion_r226835878). "Just a partial specialization" correctly avoids reversing a `reverse_view`, so `ranges::equal(x, view::reverse(view::reverse(x)))` as desired, but then additional applications of `reverse` also do nothing which is *not* desired: `ranges::equal(x, view::reverse(view::reverse(view::reverse(x))))` holds.

Reply:
The "sometimes `foo_view` does nothing" design seems more distasteful than the "`foo_view` and `view::foo` have different behaviors" design.

Reply:
Wouldn't `reverse_view<reverse_view<R>>` just return `R`'s iterator/sentinel? If `R` happens to be a `reverse_view`, didn't the right thing just happen?

Reply:
*sigh* I'm going to take a month of vacation after Ranges gets merged - I'm cooked.

Reply:
🤣

Reply:
Fixed in #931.